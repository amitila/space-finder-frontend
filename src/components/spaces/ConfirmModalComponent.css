.modal {
    position: fixed;
    z-index: 1;
    padding-top: 100px;
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
    overflow: auto;
    opacity: 1;
    background-color: antiquewhite;
}

.modal-content {
    background-color: aliceblue;
    margin: auto;
    padding: 20px;
    border: 1px solid rgb(239, 47, 47);
    width: 80%;
}

.modalText {

}