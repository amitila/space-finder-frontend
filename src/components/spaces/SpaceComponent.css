img {
    max-width: 50%;
    max-height: 50%;
    width: 150px;
    height: 150px;
    position: relative;
}

.spaceComponent {
    display: inline-block;
    width: 300px;
    margin: 20px 10px;
    overflow: auto;
}

.name {
    text-align: center;
    color: green;
    text-transform: uppercase;
}

.spaceId {

}

.location {

}