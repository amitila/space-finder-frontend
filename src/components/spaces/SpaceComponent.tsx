import { Component } from "react";
import "./SpaceComponent.css";
import amiImg from "../../assets/hcmut.png";

interface SpaceComponentProps {
  spaceId: string;
  name: string;
  location: string;
  photoUrl?: string;
  reserveSpace: (SpaceId: string) => void;
}

export class SpaceComponent extends Component<SpaceComponentProps> {
  private renderImage() {
    if (this.props.photoUrl) {
      return <img src={this.props.photoUrl} alt="img" />;
    } else {
      return <img src={amiImg} alt="ami" />;
    }
  }

  render() {
    return (
      <div className="spaceComponent">
        {this.renderImage()}
        <br />
        <label className="name">{this.props.name}</label>
        <br />
        <label className="spaceId">{this.props.spaceId}</label>
        <br />
        <label className="location">{this.props.location}</label>
        <br />
        <button onClick={() => this.props.reserveSpace(this.props.spaceId)}>
          Reverse
        </button>
      </div>
    );
  }
}
