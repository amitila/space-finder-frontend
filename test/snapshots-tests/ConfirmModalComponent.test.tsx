import React from "react";
import { create } from "react-test-renderer";
import { ConfirmModalComponent } from "../../src/components/spaces/ConfirmModalComponent";

describe("Confirm modal snapshot testing", () => {
  const closeMock = jest.fn();

  test("initial test", () => {
    const snap = create(
      <ConfirmModalComponent
        show={true}
        content={"show content"}
        close={closeMock}
      />
    );
    expect(snap).toMatchSnapshot();
  });
});
