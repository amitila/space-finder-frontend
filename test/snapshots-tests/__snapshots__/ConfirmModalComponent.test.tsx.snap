// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm modal snapshot testing initial test 1`] = `
<div
  className="modal"
>
  <div
    className="modal-content"
  >
    <h2>
      You tried to reserve and ...
    </h2>
    <h3
      className="modalText"
    >
      show content
    </h3>
    <button
      className="btnClose"
      onClick={[Function]}
    >
      Ok, close
    </button>
  </div>
</div>
`;
